compiler/circle-mpqsolver/src/pattern/PatternResolver.cpp

```diff
@@ -163,6 +163,100 @@ bool LayerNormPattern::matched()
   return true;
 }
 
+/**
+ * SUBGRAPH PATTERN
+ *  SoftmaxPattern := (exp(ifm - max(ifm))) / sum(exp(ifm - max(ifm)))
+ *    - Below diagram shows Softmax pattern to quantize.
+ *
+ *         [In]    [CircleConst] = [CircleConst(=-1)]
+ *          | \          /           |
+ *          |  \        /            |
+ *          | [CircleReduceMax]      |
+ *          |    /                   |
+ *          |   /                    |
+ *          |  /                     |
+ *        [Sub]                      |
+ *          |                        |
+ *          |                        |
+ *        [Exp]                      |
+ *          | \                      |
+ *          |  \                     |
+ *          |  [CircleSum]-----------+
+ *          |  /
+ *          | /
+ *        [Div]
+ *          |
+ *          |
+ *      [CircleNode]
+ */
+class SoftmaxPattern final
+{
+public:
+  SoftmaxPattern(luci::CircleDiv *candidate)
+  {
+    assert(candidate);
+    div_as_terminal = candidate;
+  }
+
+public:
+  bool matched();
+
+  std::vector<luci::CircleNode *> get_q16_nodes() { return {sub, exp}; }
+
+  std::vector<luci::CircleNode *> get_q8_nodes() { return {ifm, max, sum, div_as_terminal}; }
+
+public:
+  luci::CircleNode *ifm = nullptr;            // input feature map
+  luci::CircleReduceMax *max = nullptr;       // = max(ifm)
+  luci::CircleSub *sub = nullptr;             // = ifm - max(ifm)
+  luci::CircleExp *exp = nullptr;             // = exp(ifm - max(ifm))
+  luci::CircleSum *sum = nullptr;             // = sum(exp(ifm - max(ifm)))
+  luci::CircleDiv *div_as_terminal = nullptr; // = exp(ifm - max(ifm)) / sum(exp(ifm - max(ifm)))
```

Comment:
it would be better to use `_` before members for better readability like `_ifm` and so one

Reply:
ok

Reply:
fixed